portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -64,11 +64,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             // Add entity to datastore
             datastore.put(taskEntity);
 
+            response.sendRedirect("/index.html");
       } else {
-          response.getWriter().println("You are not logged in.");
+          response.sendError(401);
```

Comment:
This is good! Check if there's a way to include a custom message with the response, so the response can include a "user not logged in" or somesuch.

Also, do you need to handle this in the client side javascript? I imagine if you're not, it'll just display a big default 401 page or something.

Reply:
Yes I will solve that tomorrow as part of a different pull request where I fix several styling issues. For now I just wanted to have the error show up.